File: seed/csharp-model/trace/src/SeedTraceClient/Commons/SinglyLinkedListValue.cs
using System.Text.Json.Serialization
using SeedTraceClient

namespace SeedTraceClient

public class SinglyLinkedListValue
{
    [JsonPropertyName("head")]
    public string? Head { get; init; }

    [JsonPropertyName("nodes")]
    public Dictionary<string, SinglyLinkedListNodeValue> Nodes { get; init; }
}

File: seed/php-model/trace/src/Traits/SinglyLinkedListValue.php
<?php

namespace Seed\Traits;

use Seed\SinglyLinkedListNodeValue;
use Seed\Core\Json\JsonProperty;
use Seed\Core\Types\ArrayType;

/**
 * @property ?string $head
 * @property array<string, SinglyLinkedListNodeValue> $nodes
 */
trait SinglyLinkedListValue
{
    /**
     * @var ?string $head
     */
    #[JsonProperty('head')]
    public ?string $head;

    /**
     * @var array<string, SinglyLinkedListNodeValue> $nodes
     */
    #[JsonProperty('nodes'), ArrayType(['string' => SinglyLinkedListNodeValue::class])]
    public array $nodes;
}
